### ISSUE TITLE
二值语义分割 dice loss替换cross_entropy后，预测结果全黑，loss有问题吗？

### ISSUE THREAD

**Issue by pauperonway** [MaybeShewill-CV/lanenet-lane-detection#399]
def _dice_plus_loss(self, prediction,  one_hot, weight_map):

         prediction = tf.cast(prediction, tf.float32)

         # Dice as according to the paper:
         softmax_of_logits = tf.nn.softmax(prediction)

         dice_numerator = tf.reduce_sum(tf.multiply(softmax_of_logits * one_hot, weight_map), axis=3)
         dice_numerator = 2.0 * tf.reduce_sum(dice_numerator, axis=0)

         dice_denominator = tf.reduce_sum(tf.multiply(softmax_of_logits, weight_map), axis=3) + \
                                tf.reduce_sum(tf.multiply(one_hot, weight_map), axis=3)
         dice_denominator = tf.reduce_sum(dice_denominator, axis=0)

         epsilon = 0.00001
         loss_dice = (dice_numerator + epsilon) / (dice_denominator + epsilon)
         return 1 - tf.reduce_mean(loss_dice)
此函数参考NiftyNet写的。求看......

**Comment by pauperonway** [comment]
在compute_loss中调用方式如下：
binary_segmenatation_loss = self._dice_plus_loss(prediction = binary_seg_logits, one_hot = binary_label_onehot,  weight_map = inverse_weights)
